polaris/viz/__init__.py

```diff
@@ -0,0 +1,233 @@
+                     cmap=None, cmap_set_under=None, cmap_scale='linear'):
+
+    """
+    Plot a horizontal field from a planar domain using x,y coordinates at a
+    single time and depth slice.
+
+    Parameters
+    ----------
+    config : polaris.config.PolarisConfigParser
+        Configuration options with parameters used to construct the vertical
+        grid
+
+    ds : xarray.Dataset
+        A data set containing fieldName
+
+    ds_mesh : xarray.Dataset
+        A data set containing mesh variables
+
+    field_name: str
+        The name of the variable to plot, which must be present in ds
+
+    out_file_name: str
+        The path to which the plot image should be written
+
+    title: str, optional
+        The title of the plot
+
+    vmin, vmax : float, optional
+        The minimum and maximum values for the colorbar
+
+    show_patch_edges : boolean, optional
+        If true, patches will be plotted with visible edges
+
+    t_index, z_index: int, optional
+        The indices of 'Time' and 'nVertLevels' axes to select for plotting
+
+    cmap : Colormap or str, optional
+        A color map to plot
+
+    cmap_set_under : str or None, optional
+        A color for low out-of-range values
+
+    cmap_scale : {'log', 'linear'}, optional
+        Whether the colormap is logarithmic or linear
+    """
+    style_path = os.path.realpath(os.path.dirname(__file__))
+    plt.style.use(f'{style_path}/polaris.mplstyle')
+
+    try:
+        os.makedirs(os.path.dirname(out_file_name))
+    except OSError:
+        pass
+
+    if os.path.exists(out_file_name):
+        return
+
+    if title is None:
+        title = field_name
+
+    if 'maxLevelCell' not in ds_mesh:
+        raise ValueError(
+            'maxLevelCell must be added to ds_mesh before plotting.')
+    if field_name not in ds:
+        raise ValueError(
+            f'{field_name} must be present in ds before plotting.')
+
+    field = ds[field_name]
+
+    if 'Time' in field.dims and t_index is None:
+        t_index = 0
+    if t_index is not None:
+        field = field.isel(Time=t_index)
+    if 'nVertLevels' in field.dims and z_index is None:
+        z_index = 0
+    if z_index is not None:
+        field = field.isel(nVertLevels=z_index)
+
+    if 'nCells' in field.dims:
+        ocean_mask = ds_mesh.maxLevelCell - 1 >= 0
+        ocean_patches, ocean_mask = _compute_cell_patches(ds_mesh, ocean_mask)
+    elif 'nEdges' in field.dims:
+        ocean_mask = np.ones_like(field, dtype='bool')
+        ocean_mask = _remove_boundary_edges_from_mask(ds_mesh, ocean_mask)
+        ocean_patches, ocean_mask = _compute_edge_patches(ds_mesh, ocean_mask)
+    ocean_patches.set_array(field[ocean_mask])
+    if cmap is not None:
+        ocean_patches.set_cmap(cmap)
+    if cmap_set_under is not None:
+        current_cmap = ocean_patches.get_cmap()
+        current_cmap.set_under(cmap_set_under)
+
+    if show_patch_edges:
+        ocean_patches.set_edgecolor('black')
```

Comment:
I think this is going to be tricky at higher res.  I tried 4k and it seems better with`'face'`:
![vel_no_edges](https://user-images.githubusercontent.com/4179064/231513889-1d79a2df-1b89-4c3d-8a08-eb84c1fc710c.png)

 than with `'black'`:
![vel_edges](https://user-images.githubusercontent.com/4179064/231513939-6ff2f6ae-4ced-4a6e-8aec-136192683546.png)

Reply:
Sounds good. I'll remove the `show_patch_edges=True` from this line so it defaults back to false here and in `initial_state`
https://github.com/E3SM-Project/polaris/blob/af735204caca8fda6116c49c3acc25ec36b4513a/polaris/ocean/tests/baroclinic_channel/viz.py#L51

Reply:
Ah, I see.  In that case, I think it's fine.  I misunderstood that it was hard-coded to showing edges for edge quantities. 
 Now, I see that was just a choice you made.